src/Microsoft.SqlTools.Hosting/Utility/SqlClientEventListener.cs

```diff
@@ -36,22 +36,25 @@ protected override void OnEventSourceCreated(EventSource eventSource)
         /// <param name="eventData">The data for the event</param>
         protected override void OnEventWritten(EventWrittenEventArgs eventData)
         {
-            if (eventData.Payload == null)
+            // Skip EventCounters as they can come from any event source and will pollute traces captured.
+            if (!eventData.EventName.Equals(nameof(EventCounter)))
```

Comment:
nit: update the if condition checking for EventCounters to return early, like the `eventData.Payload == null` check below, for easier readability